### ISSUE TITLE
Jsdoc Type alias circularly references itself

### ISSUE THREAD

**Issue by awerlogus** [microsoft/TypeScript#39372]
**TypeScript Version:**  4.0.0-dev.20200701

**Search Terms:** Jsdoc type alias circularly references itself

I'm trying to describe json type in `.js` file using jsdoc.

```js
// Type alias 'JsonArray' circularly references itself.
/** @typedef {ReadonlyArray<Json>} JsonArray */
/** @typedef {{ readonly [key: string]: Json }} JsonRecord */
// Type alias 'Json' circularly references itself.
/** @typedef {boolean | number | string | null | JsonRecord | JsonArray | readonly []} Json */
```

It shows me some errors that JsonArray and Json types circularly reference to itself. 

Tried the same in `.ts` file. There got no errors.

```ts
type JsonArray = ReadonlyArray<Json>
type JsonRecord = { readonly [key: string]: Json }
// OK
type Json = boolean | number | string | null | JsonRecord | JsonArray | readonly []
```

Expected behaviour: Json types in `.js` and `.ts` files should work identical.

**Comment by sandersn** [comment]
Note that this didn't work in Typescript prior to 3.7, so it might be just a matter of making sure those changes work for JS.

**Comment by sandersn** [comment]
There is code in getAliasSymbolForTypeNode for `@typedef`, but it doesn't work: it doesn't skip past the typedef type expression correctly.

**Comment by jcorbin** [comment]
When will this fix be released? Will there be a 4.0.4 patch, or must we wait for 4.1 minor?

**Comment by sandersn** [comment]
It's not in 4.0.5, so it'll only be in 4.1 and above.
